data-workflows/plugin/helpers.py

```diff
@@ -0,0 +1,157 @@
+        s3_client = boto3.client("s3")
+        return json.loads(s3_client.get_object(Bucket=bucket, Key=os.path.join(bucket_path, key))['Body'].read())
+    except ClientError:
+        logging.info(f"Not cached: {key}")
+        return None
+
+
+def get_valid_plugins() -> Dict[str, str]:
+    return {**_get_cache('cache/hidden-plugins.json'), **_get_cache('cache/public-plugins.json')}
+
+
+def get_frontend_manifest_metadata(plugin, version):
+    raw_metadata = get_manifest(plugin, version)
+    if 'error' in raw_metadata:
+        raw_metadata = None
+    interpreted_metadata = parse_manifest(raw_metadata)
+    return interpreted_metadata
+
+
+def parse_manifest(manifest: Optional[dict] = None):
+    """
+    Convert raw manifest into dictionary of npe2 attributes.
+    :param manifest: raw manifest
+    """
+    manifest_attributes = {
+        'display_name': '',
+        'plugin_types': [],
+        'reader_file_extensions': [],
+        'writer_file_extensions': [],
+        'writer_save_layers': [],
+    }
+    if manifest is None:
+        return manifest_attributes
+    manifest_attributes['display_name'] = manifest.get('display_name', '')
+    manifest_attributes['npe2'] = not manifest.get('npe1_shim', False)
+    if 'contributions' in manifest:
+        manifest_contributions = manifest['contributions']
+        if 'readers' in manifest_contributions:
+            readers = manifest_contributions['readers']
+            if readers:
+                manifest_attributes['plugin_types'].append('reader')
+                reader_file_extensions = set()
+                for reader in readers:
+                    filename_patterns = reader.get('filename_patterns', [])
+                    for ext in filename_patterns:
+                        reader_file_extensions.add(ext)
+                manifest_attributes['reader_file_extensions'] = list(reader_file_extensions)
+        if 'writers' in manifest_contributions:
+            writers = manifest_contributions['writers']
+            if writers:
+                manifest_attributes['plugin_types'].append('writer')
+                writer_file_extensions = set()
+                writer_save_layers = set()
+                for writer in writers:
+                    filename_extensions = writer.get('filename_extensions', [])
+                    layer_types = writer.get('layer_types', [])
+                    for ext in filename_extensions:
+                        writer_file_extensions.add(ext)
+                    for layer_type in layer_types:
+                        if match := re.match(VALID_LAYER_REGEX, layer_type):
+                            writer_save_layers.add(match.groups()[0])
+                manifest_attributes['writer_file_extensions'] = list(writer_file_extensions)
+                manifest_attributes['writer_save_layers'] = list(writer_save_layers)
+        if 'themes' in manifest_contributions and manifest_contributions['themes']:
```

Comment:
```suggestion
        if manifest_contributions.get('themes'):
```

and similar down the chain